### ISSUE TITLE
Internationalization

### ISSUE THREAD

**Issue by headzoo** [DeskproApps/deskpro-components#42]
If we plan to support i18n (and we should) now seems like the time to add it. Even if there's no plan to translate messages, i18n is still needed to format numbers and dates.

**Comment by jducro** [comment]
I completely agree on that, and the earlier the better and I know how can date and number format can be different in another language 😉

**Comment by headzoo** [comment]
It probably won't even be that hard, since most string translating would happen at a level above deskpro-components, e.g. the DeskPRO app. We only need localization for numbers, possibly dates, and a few simple messages like "Open", "Close", and stuff like that.

**Comment by jducro** [comment]
We already use https://github.com/yahoo/react-intl in Deskpro somewhere. Did you have something else in mind?

**Comment by headzoo** [comment]
I think if we have an eye on potentially open sourcing deskpro-components, we have to be mindful about forcing the end users to adopt a specific i18n library, or having one tightly integrated with our components.

After a quick glance through the components, we only need to translate the following messages:

* 'Are you sure?'
* 'Please select'
* 'Add a label'
* 'Close'
* 'Open'

The simplest approach would be making those message configurable via props, and letting the end user translate them using their i18n library of choice. That leaves us with only having to "translate" numbers and dates. We could do that easily with some kind of global configuration. For example:

```jsx
import Config from 'deskpro-components/config';
Config.setLocale('de');
```

Any function which formats numbers/dates can read from that config.

To format numbers we could have a simple json file which maps locales to thousandths place separators (dot, comma, or space) and localization of dates is easily handled by [react-moment](https://github.com/headzoo/react-moment).

**Comment by jducro** [comment]
It's already the case for 'Add a label', 'Please select' and 'Are you sure?' they are only set as defaultProps.

Agree on using moment for dates.

I guess the config should allowed to specify the locale and also to specifically overwrite thousandths separator and decimal separator (it's a comma in French ;) )

**Comment by headzoo** [comment]
I guess the real question is... do we want to have a global config registry? I don't have a problem with adding one. There may be some issues with testing, but they can be handled.
